### DISCUSSION TITLE
Run script inside container from script on host

### DISCUSSION THREAD

**Discussion by cjnewbs** [wardenenv/warden#654] [Help & Support] [answered]
Hi,
I'm creating a makefile to setup local environments automatically.
I've got most of it working but I am having trouble running `bin/magento s:up`. I can run it interactively from within `warden shell` but I would like to be able to do something like:
```
echo "bin/magento s:up" | warden shell
```
Which is similar to running a setup SQL which I do like so:`echo "$$SQL" | warden db import`, however trying it with `warden shell` results in the error `the input device is not a TTY`

Any ides?

**Comment by bap14** [comment, marked as answer]
@cjnewbs You should be able to do `warden shell -c "bin/magento setup:upgrade"`. Since the `warden shell` command is just executing the bash command in the container, you can use normal bash flags and parameters after the `warden shell` call.

**Reply by cjnewbs** [reply]
This does exactly what I want, thank you!

**Reply by navarr** [reply]
Thanks bap14.  I wonder how difficult it would be to implement piping into the command, though. 🤔
